tests/test_external_roundtrip.rs

```diff
@@ -0,0 +1,116 @@
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+extern crate tempdir;
+
+use tempdir::TempDir;
+
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+/// A macro used to simplify debugging.
+const DEBUG: bool = true;
+
+fn get_path(root: &Path, components: &[&str]) -> PathBuf {
+    let mut root = root.to_path_buf();
+    for component in components {
+        root.push(component)
+    }
+    root
+}
+
+fn run(mut command: std::process::Command) -> Result<(), std::io::Error> {
+    let output = command.output()?;
+    if DEBUG || !output.status.success() {
+        debug!(target: "test", "Command: {command:?}\n** out:{out}\n** err:{err}",
+            command = command,
+            out =  String::from_utf8(output.stdout).unwrap_or_else(|_| "<cannot decode output>".to_string()),
+            err =  String::from_utf8(output.stderr).unwrap_or_else(|_| "<cannot decode output>".to_string())
+        );
+    }
+    assert!(output.status.success());
+    Ok(())
+}
+
+fn perform_test(root: &Path, prefix: &str, source: &str) {
```

Comment:
Hmm, my other comment from prev review seems to got lost.

I suggested to join `root` + `prefix` just once at the top, and then also use `.join` for all files below instead of custom util with array. Something like:

```rust
let root = root.join(prefix);
...
let path_tmp_source = root.join("sample.js");
let path_out_encoded_1 = root.join("encoded_1");
let path_in_encoded_1 = path_out_encoded_1.join("sample.binjs");
...
```